File: glasses-management-backend-h2/src/main/java/com/glasses/entity/Customer.java
package com.glasses.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.util.Date;

@Data
@TableName("customer")
public class Customer {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String phone;
    private Integer gender; // 1 male, 2 female, 0 unknown

    @JsonFormat(pattern = "yyyy-MM-dd", timezone = "Asia/Shanghai")
    private Date birthday;
    private String remark;
    private Boolean deleted;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date deletedTime;
    private Long deletedBy;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date createTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date updateTime;
}

File: glasses-management-android-native/app/src/main/java/com/glasses/native/data/model/Customer.kt
package com.glasses.native.data.model

data class Customer(
    val id: Long? = null,
    val name: String,
    val phone: String? = null,
    val gender: Int = 0,
    val birthday: String? = null,
    val remark: String? = null,
    val deleted: Boolean? = null,
    val deletedTime: String? = null,
    val deletedBy: Long? = null,
    val createTime: String? = null,
    val updateTime: String? = null
)

data class CustomerPageResponse(
    val records: List<Customer>,
    val total: Long,
    val size: Int,
    val current: Int,
    val pages: Int
)
